### PR TITLE
feat: Initial implementation of active deployment action

### PR DESCRIPTION
**Pull request by snorremd**
Implement action that finds the nth most recent active deploy in a GitHub repository. Defaults to finding the first deployment. If nth is 2 it will find the second most recent deployment.

Allows users to override which repository (owner and repo) to fetch deployments from. Defaults to the one in the github.context.

Will page through deployments with a slight politeness delay to avoid spamming GitHub API.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,89 +1,66 @@
...
-| Name           | Description                          | Required | Default |
-| -------------- | ------------------------------------ | -------- | ------- |
-| `github-token` | GitHub token                         | true     |         |
-| `workspace`    | Turbo workspace name                 | true     |         |
-| `prefix`       | Prefix for release title             | false    | ""      |
-| `from`         | Commit SHA to start from (exclusive) | true     |         |
-| `to`           | Commit SHA to end at                 | true     |         |
+| Name           | Description             | Required | Default                   |
+| -------------- | ----------------------- | -------- | ------------------------- |
+| `github-token` | GitHub token            | true     |                           |
```
Comment (boyum) [inline README.md:28]: Is it required, or will it use the default GitHub token as default?

File: README.md [hunk 2]
```diff
@@ -1,89 +1,66 @@
...
+| `environment`  | Deployment environment  | true     |                           |
+| `owner`        | GitHub repository owner | false    | github.context.repo.owner |
+| `repo`         | GitHub repository name  | false    | github.context.repo.repo  |
```
Comment (boyum) [inline README.md:31]: Not sure if the technical details are important here?

```suggestion
| `owner`        | GitHub repository owner | false    | Current repo's owner |
| `repo`         | GitHub repository name  | false    | Current repo  |
```

File: src/helpers.ts [hunk 3]
```diff
@@ -1,243 +1,148 @@
...
-  const relevantCommits: GitLog[] = []
-
-  // Checkout commit using shell script
-  for (const commit of commits) {
-    const checkout = await exec('git', ['checkout', commit.sha])
-    if (checkout !== 0) {
-      continue
+  environment: string,
+  nth: number
+): Promise<number | null> {
```
Comment (boyum) [inline src/helpers.ts:103]: For the future, is there a way to cache this? If we see a previously checked commit and still no new deployments, just returned the previous result?

```suggestion
): Promise<number | null> {
  // TODO: Return previous result if we find a commit that's been checked previously
  // and there are still no new commits
```

File: src/helpers.ts [hunk 4]
```diff
@@ -1,243 +1,148 @@
...
-            result += data.toString()
-          }
-        }
-      }
-    )
 
-    if (exitCode !== 0) {
-      continue
+    const deployments = await fetchDeployments(octokit, context, environment, 20, cursor)
+    for(const deployment of deployments.nodes.filter(d => d.state === 'ACTIVE')) {
```
Comment (boyum) [inline src/helpers.ts:114]: TODO: Use prettier :)

File: README.md [hunk 5]
```diff
@@ -1,89 +1,66 @@
...
+| `nth`          | Nth deployment          | false    | 1                         |
```
Comment (boyum) [inline README.md:32]:
```suggestion
| `nth`          | Nth deployment (1 based)         | false    | 1                         |
```

**Review by boyum — COMMENTED** [review]
Looks really good! I love how readable the code is and the 100% test coverage 🤩 I'll definitely use this as inspiration in the future.

File: src/helpers.ts (hunk 3 above)
Comment (snorremd) [reply src/helpers.ts:104]: Deployments and releases can be deleted by users, so re-running is the safest bet. Optional caching could perhaps be something to look into. Normally it doesn't have to page that far.

File: README.md [hunk 6]
```diff
@@ -1,89 +1,65 @@
...
-| Name           | Description                          | Required | Default |
-| -------------- | ------------------------------------ | -------- | ------- |
-| `github-token` | GitHub token                         | true     |         |
-| `workspace`    | Turbo workspace name                 | true     |         |
-| `prefix`       | Prefix for release title             | false    | ""      |
-| `from`         | Commit SHA to start from (exclusive) | true     |         |
-| `to`           | Commit SHA to end at                 | true     |         |
+| Name           | Description             | Required | Default                             |
+| -------------- | ----------------------- | -------- | ----------------------------------- |
+| `github-token` | GitHub token            | true     | Defaults to the github action token |
```
Comment (boyum) [inline README.md:27]:
```suggestion
| `github-token` | GitHub token            | false    | Defaults to the github action token |
```

File: action.yml [hunk 7]
```diff
@@ -15,33 +15,31 @@ inputs:
   github-token:
     description: 'A github token used to create the release'
     required: true
```
Comment (boyum) [inline action.yml:17]:
```suggestion
    required: false
```

**Review by boyum — APPROVED** [review]

**Pull request merged**